compiler/resolution/resolveFunction.cpp

```diff
@@ -567,6 +567,7 @@ static void insertUnrefForArrayOrTupleReturn(FnSymbol* fn) {
           VarSymbol* tmp       = newTemp(arrayUnrefName, rhsType);
           CallExpr*  initTmp   = new CallExpr(PRIM_MOVE,     tmp, rhs);
           CallExpr*  unrefCall = new CallExpr("chpl__unref", tmp);
+          CallExpr*  shapeSet  = toCallExpr(call->next);
```

Comment:
I really don't like relying on AST patterns with elements that happen to be next to each other. What if error handling (say) adds another call in between? Is there some reason you know such things can't happen? Is there an alternative representation or way to find the relevant primitive?

Reply:
Good point. We could have a hash table from each move into the ret symbol to the corresponding shape thing. (There could be several such moves per function, ex. if the user has multiple return statements.) This way of tracking dependencies would come handy in other cases as well. We are not set up for it right now because such a hash table would need to be updated whenever a function - or its part - is cloned.

Instead, how about I write a little helper that traverses all statements from `call->next` on, until it finds a PRIM_ITERATOR_RECORD_SET_SHAPE ?

Reply:
I'm OK with the helper, yeah